src/m365/spo/commands/listitem/listitem-roleassignment-add.ts

```diff
@@ -0,0 +1,271 @@
+import { Cli, CommandOutput, Logger } from '../../../../cli';
+import Command, { CommandErrorWithOutput } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { formatting, urlUtil, validation } from '../../../../utils';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+import * as SpoUserGetCommand from '../user/user-get';
+import { Options as SpoUserGetCommandOptions } from '../user/user-get';
+import * as SpoGroupGetCommand from '../group/group-get';
+import { Options as SpoGroupGetCommandOptions } from '../group/group-get';
+import * as SpoRoleDefinitionListCommand from '../roledefinition/roledefinition-list';
+import { Options as SpoRoleDefinitionListCommandOptions } from '../roledefinition/roledefinition-list';
+import { RoleDefinition } from '../roledefinition/RoleDefinition';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listItemId: number;
+  listId?: string;
+  listTitle?: string;
+  listUrl?: string;
+  principalId?: number;
+  upn?: string;
+  groupName?: string;
+  roleDefinitionId?: number;
+  roleDefinitionName?: string;
+}
+
+class SpoListItemRoleAssignmentAddCommand extends SpoCommand {
+  public get name(): string {
+    return commands.LISTITEM_ROLEASSIGNMENT_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a role assignment to listitem permissions';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        listId: typeof args.options.listId !== 'undefined',
+        listTitle: typeof args.options.listTitle !== 'undefined',
+        listUrl: typeof args.options.listUrl !== 'undefined',
+        principalId: typeof args.options.principalId !== 'undefined',
+        upn: typeof args.options.upn !== 'undefined',
+        groupName: typeof args.options.groupName !== 'undefined',
+        roleDefinitionId: typeof args.options.roleDefinitionId !== 'undefined',
+        roleDefinitionName: typeof args.options.roleDefinitionName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '--listItemId <listItemId>'
+      },
+      {
+        option: '-i, --listId [listId]'
+      },
+      {
+        option: '-t, --listTitle [listTitle]'
```

Comment:
Let's remove both these short options as they were also removed in the specs.

Reply:
same comment about the shorts. I think it is more or less aligned with rest of the commands in this namespace 🤔. Maybe we could double check it or if there is some inconsistent  approach we could do it as a separate issue 🤔

Reply:
When reading through the assigned issue I noticed this comment https://github.com/pnp/cli-microsoft365/issues/3539#issuecomment-1221376971 😄

Reply:
ok that got me convinced as well 😉